src/chemenv/tools/clinical_trials.py

```
@@ -0,0 +1,437 @@
+                    if response.status == 429:
+                        retry_after = int(response.headers.get("Retry-After", 60))
+                        logger.warning(
+                            f"Rate limited. Retrying after {retry_after} seconds."
+                        )
+                        await asyncio.sleep(retry_after)
+                        raise aiohttp.ClientResponseError(
+                            request_info=response.request_info,
+                            history=response.history,
+                            status=429,
+                            message="Rate limited",
+                        )
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)